core/dbt/contracts/graph/compiled.py

```diff
@@ -109,15 +109,19 @@ class CompiledSnapshotNode(CompiledNode):
 @dataclass
 class CompiledDataTestNode(CompiledNode):
     resource_type: NodeType = field(metadata={'restrict': [NodeType.Test]})
-    config: TestConfig = field(default_factory=TestConfig)
+    # Was not able to make mypy happy and keep the code working. We need to
+    # refactor the various configs.
+    config: TestConfig = field(default_factory=TestConfig)  # type:ignore
```

Comment:
It looks like your suggestion to remove TestConfig as a subclass of NodeConfig would fix this, but it looks like we'd have to stop pulling that information from `field` and use a function that mypy can follow the types of.

Can we update the comment to reference the problem / possible fix here just so the information doesn't get lost?

Reply:
Ahh gotcha. I played around with it myself a bit and got it wrong. Definitely fine to push off.